lib/tenstorrent/boot_fs/tt_boot_fs.c

```diff
@@ -175,3 +177,90 @@ int tt_boot_fs_get_file(const tt_boot_fs *tt_boot_fs, const uint8_t *tag, uint8_
 
 	return TT_BOOT_FS_OK;
 }
+
+int tt_boot_fs_ls(const struct device *dev, tt_boot_fs_fd *fds, size_t nfds, size_t offset)
+{
+	if (!dev || !device_is_ready(dev)) {
+		return -ENXIO;
+	}
+
+	tt_boot_fs_fd temp_fds[CONFIG_TT_BOOT_FS_IMAGE_COUNT_MAX];
+
+	if (flash_read(dev, TT_BOOT_FS_FD_HEAD_ADDR, (uint8_t *)temp_fds, sizeof(temp_fds)) != 0) {
+		return -EIO;
+	}
+
+	if (offset >= CONFIG_TT_BOOT_FS_IMAGE_COUNT_MAX) {
+		return -EINVAL;
+	}
+
+	size_t count = 0;
+	size_t found = 0;
+
+	/* Iterate through the fds now in RAM, starting from offset */
+	for (size_t i = 0; i < CONFIG_TT_BOOT_FS_IMAGE_COUNT_MAX; i++) {
+		tt_boot_fs_fd *current_fd = &temp_fds[i];
+
+		if (current_fd->flags.f.invalid) {
+			break;
+		}
+
+		tt_checksum_res_t chk_res = calculate_and_compare_checksum(
+			(uint8_t *)current_fd, sizeof(tt_boot_fs_fd) - sizeof(uint32_t),
+			current_fd->fd_crc, false);
+
+		if (chk_res == TT_BOOT_FS_CHK_OK) {
+			found++;
+		}
+	}
+
+	/* If no valid descriptors found at all, treat as not a valid boot fs */
+	if (found == 0 && offset == 0) {
+		return -ENXIO;
+	}
```

Comment:
If any invalid file descriptor is found (aside from a possible terminating FD?) then we should probably return an error.

```suggestion
```